### PR TITLE
internal: move the setup_code as init_code into ElementRc

### PR DESCRIPTION
**Pull request by tronical**
This also matches the `init_code` naming in the llr component and prepares for allowing user code in that code block.

### CODE REVIEW

File: internal/compiler/object_tree.rs [hunk 1]
```diff
@@ -469,6 +466,9 @@ pub struct Element {
     pub bindings: BindingsMap,
     pub property_analysis: RefCell<HashMap<String, PropertyAnalysis>>,
 
+    /// Code to be inserted into the constructor for root elements of components
```
Comment (ogoffart) [inline internal/compiler/object_tree.rs:469]: Why do we actually restrict of root elements?
Also, within repeater, it is not the root element.  (even if a "component" will be created later, this is not conceptually a component)

Comment (tronical) [reply internal/compiler/object_tree.rs:469]: That's true (the repeaters part). I mean, yeah, we could also allow `init => {}` anywhere. I'll adjust the comment in this PR to describe the reality, and we can discuss the new scope in the other PR.

**Review by ogoffart — APPROVED** [review]
Not a strongly held opinion, but i think it would be easier to keep the init code in a binding for a imaginary `init` callback, and lower that to the `setup_code` in the Component later.  (the same way we keep `opacity` or `visible` as binding then lower that to an element in the passes.)

**Comment by tronical** [comment]
> Not a strongly held opinion, but i think it would be easier to keep the init code in a binding for a imaginary `init` callback, and lower that to the `setup_code` in the Component later. (the same way we keep `opacity` or `visible` as binding then lower that to an element in the passes.)

That's fine with me, too. Might indeed be easier. In that case this PR isn't needed :)

**Pull request closed without merging**